rfc.md

```diff
@@ -0,0 +1,322 @@
+
+### /get_people 
+Returns a list of people
+    
+#### Success Response (Without a body)
+```json
+{
+    "status": {
+        "messages": [
+            "People successfully found"
+        ],
+        "success": true
+    },
+    "people": [
+      {"uuid": "<UUID>", "name": "<NAME>"},
+      {"uuid": "<UUID>", "name": "<NAME>"},
+      ...
+    ]
+}
+```
+
+### /get_person 
+returns a person, including uuid
+
+#### BODY
+```json
+{
+  "name": "The person's name"
+}
+```
+
+#### Success Response
+```json
+{
+  "status": {
+    "messages": [
+      "Person was retrieved successfully"
+    ],
+    "success": true
+  },
+  "person": {
+    "uuid": "<UUID>",
+    "name": "<NAME>"
+  }
+}
+```
+
+#### Not Found Response
+```json
+{
+  "status": {
+    "messages": [
+      "No person with that name could be found"
+    ],
+    "success": false
+  }
+}
+```
+
+### /create_person 
+creates a person and returns that object with a UUID
+
+#### BODY
+```json
+{
+  "name": "The name of the person to create"
+}
+```
+
+#### Success Response
+```json
+{
+  "status": {
+    "messages": [
+      "Entry for person was successfully created"
+    ],
+    "success": true
+  },
+  "person": {
+    "uuid": "<UUID>",
+    "name": "<NAME>"
+  }
+}
+```
+
+#### Error Response (Already Exists)
+```json
+{
+  "status": {
+    "messages": [
+      "Cannot create person, someone with that name already exists."
+    ],
+    "success": false
+  }
+}
+```
+
+### /delete_person 
+marks a person and their friendships as removed
+
+#### BODY
+```json
+{
+  "uuid": "The person who you are deleting's uuid"
+}
+```
+#### Successful Response
+```json
+{
+  "status": {
+    "messages": [
+      "Person successfully removed"
+    ],
+    "success": true
+  }
+}
+```
+#### Error Response (Person doesn't exist)
+```json
+{
+  "status": {
+    "messages": [
+      "No person with that id could be found"
+    ],
+    "success": false
+  }
+}
+```
+
+#### /get_friends 
+returns a list of that person's friends
+
+#### BODY
+```json
+  {
+    "uuid": "The person whose friends you are querying's UUID"
+  }
+```
+    
+#### Successful Response
+```json
+{
+  "status": {
+    "messages": [
+      "Person successfully found"
+    ],
+    "success": true
+  },
+  "person": {
+    "uuid": "<UUID>",
+    "name": "<NAME>"
+  },
+  "friends": [
+    {
+      "uuid": "<UUID>",
+      "name": "<NAME>"
+    },
+    {
+      "uuid": "<UUID>",
+      "name": "<NAME>"
+    },
+    ...
+  ]
+}
+```
+
+#### Error Response (person cannot be found)
+```json
+{
+  "status": {
+    "messages": [
+      "Person cannot be found"
+    ],
+    "success": false
+  }
+}
+```
+
+### /create_friendship 
+creates a friendship
+
+#### BODY
+```json
+{
+  "uuids": ["first UUID", "second UUID"]
+}
+```
+
+#### Successful Response
+```json
+{
+  "status": {
+    "messages": [
+      "Friendship successfully created"
+    ],
+    "success": true
+  },
+  "person": {
```

Comment:
I don't think initially we need to return person information when creating a friendship. Let's just return a status. Later we can add this as an enhancement.